src/main/java/com/google/firebase/auth/AbstractFirebaseAuth.java

```diff
@@ -63,17 +63,13 @@
       final FirebaseApp firebaseApp,
       Supplier<FirebaseTokenFactory> tokenFactory,
       Supplier<? extends FirebaseTokenVerifier> idTokenVerifier,
-      Supplier<? extends FirebaseTokenVerifier> cookieVerifier) {
+      Supplier<? extends FirebaseTokenVerifier> cookieVerifier,
+      Supplier<FirebaseUserManager> userManager) {
```

Comment:
Can we shift the `FirebaseAuth.Builder` into this class, and reuse it in both child classes? A constructor with this many args is a little strange.

Reply:
I've made this change, but I have to admit, I find this design a little strange.

Fortunately, this design is not exposed to any users, so I suppose it can always be iterated upon in the future.